jax/experimental/tf_bridge/tests/stax_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from absl.testing import absltest
+import jax
+from jax import test_util as jtu
+import numpy as np
+
+from jax.experimental import tf_bridge
+
+from jax.experimental import stax
+from jax.experimental.stax import (AvgPool, BatchNorm, Conv, Dense, FanInSum,  # pylint: disable=g-multiple-import
+                                   FanOut, Flatten, GeneralConv, Identity,
+                                   MaxPool, Relu, LogSoftmax)
+
+from jax.config import config
+config.parse_flags_with_absl()
+
+
+def ConvBlock(kernel_size, filters, strides=(2, 2)):  # pylint: disable=invalid-name
+  ks = kernel_size
+  filters1, filters2, filters3 = filters
+  Main = stax.serial(  # pylint: disable=invalid-name
+      Conv(filters1, (1, 1), strides), BatchNorm(), Relu,
+      Conv(filters2, (ks, ks), padding='SAME'), BatchNorm(), Relu,
+      Conv(filters3, (1, 1)), BatchNorm())
+  Shortcut = stax.serial(Conv(filters3, (1, 1), strides), BatchNorm())  # pylint: disable=invalid-name
+  return stax.serial(FanOut(2), stax.parallel(Main, Shortcut), FanInSum, Relu)
+
+
+def IdentityBlock(kernel_size, filters):  # pylint: disable=invalid-name
+  ks = kernel_size
+  filters1, filters2 = filters
+  def make_main(input_shape):
+    # the number of output channels depends on the number of input channels
+    return stax.serial(
+        Conv(filters1, (1, 1)), BatchNorm(), Relu,
+        Conv(filters2, (ks, ks), padding='SAME'), BatchNorm(), Relu,
+        Conv(input_shape[3], (1, 1)), BatchNorm())
+  Main = stax.shape_dependent(make_main)  # pylint: disable=invalid-name
+  return stax.serial(FanOut(2), stax.parallel(Main, Identity), FanInSum, Relu)
+
+
+def ResNet50(num_classes):  # pylint: disable=invalid-name
```

Comment:
I'm a bit sad at having multiple copies of the Stax model around, especially given we aren't really maintaining Stax in favor of other libraries.

Reply:
This was originally using Flax, but I think that the Flax + jax_to_tf should go into the Flax repo. As a substitute, Tom wrote this. I think it is fine as a test, for now.

Reply:
Given that we're inside JAX core now we could just `from jax.examples import resnet50` and replace this all with `resnet50.ResNet50(..)`?

Reply:
Done.